prov/gni/src/gnix_ep.c

```diff
@@ -1538,17 +1594,26 @@ int gnix_ep_open(struct fid_domain *domain, struct fi_info *info,
 	*ep = &ep_priv->ep_fid;
 	return ret;
 
-err1:
-	__fr_freelist_destroy(ep_priv);
 err:
+	if (free_list_inited == true)
+		__fr_freelist_destroy(ep_priv);
+
 	if (ep_priv->vc_ht != NULL) {
 		_gnix_ht_destroy(ep_priv->vc_ht); /* may not be initialized but
 						     okay */
 		free(ep_priv->vc_ht);
 		ep_priv->vc_ht = NULL;
 	}
-	if (ep_priv->cm_nic != NULL)
+	if (ep_priv->cm_nic != NULL) {
 		ret = _gnix_cm_nic_free(ep_priv->cm_nic);
+		ep_priv->cm_nic = NULL;
```

Comment:
Is this NULL assignment necessary?  Doesn't seem like it to me, so I want my 1 instruction back!